### ISSUE TITLE
make with pattern targets do not autocomplete

### ISSUE THREAD

**Issue by miki725** [fish-shell/fish-shell#7326]
It looks like if the makefile has any targets with a pattern do not autocomplete.

Here is a simple makefile without any patterns:

```
❯ cat Makefile
foo:
	echo foo
bar:
	echo bar

❯ make <TAB>
bar  (Target)  foo  (Target)
```

But if a simple pattern is added, it no longer works:

```
❯ cat Makefile
foo:
	echo foo
bar:
	echo bar
baz%:
	echo $*

❯ make <TAB>
```

Not sure if I need to enable something for `fish` to autocomplete there makefiles.

version info:

```
❯ fish --version
fish, version 3.1.2

❯ echo $version
3.1.2

❯ uname -a
Darwin MacBook-Pro.local 19.6.0 Darwin Kernel Version 19.6.0: Thu Jun 18 20:49:00 PDT 2020; root:xnu-6153.141.1~1/RELEASE_X86_64 x86_64 i386 MacBookPro15,1 Darwin

❯ echo $TERM
xterm-256color
```

same behavior is exhibited even if I try the same test without any fish configs.

```
sh -c 'env HOME=$(mktemp -d) fish'
```

**Comment by faho** [comment]
Please post the output of the following in that context:

```fish
    if make --version 2>/dev/null | string match -q 'GNU*'
        make -pRrq :
    else
        make  -d g1 -rn
    end
```

**Comment by miki725** [comment]
thanks for the tip. so its definitely related to `make` version somehow.

I use linux and mac and sometimes outdated mac version of tools were annoying to deal with due to slightly diff flags. `brew` installs gnu versions of tools but sometimes it does not symlink it under standard tool name like `make` but links it with `g*` prefix like `gmake`. since some of my scripts use directly `make`, I simply added directory of where homebrews `make` is installed to `PATH` hence essentially making `gmake` default alias for `make` on a mac. looks like that is causing the issue. to replicate it simplified the env somewhat:

with default mac's make it works:

```
❯ echo $PATH
/usr/local/bin /usr/local/sbin /Users/miki725/.fzf/bin /usr/bin /bin /usr/sbin /sbin

❯ cat Makefile
foo:
	echo foo
bar:
	echo bar
baz%:
	echo $*

❯ make --version
GNU Make 3.81
Copyright (C) 2006  Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.

This program built for i386-apple-darwin11.3.0

❯ make
bar  (Target)  foo  (Target)
```

but if I add the gnu make to path it no longer works:

```
❯ set -gx PATH /usr/local/opt/make/libexec/gnubin $PATH

❯ echo $PATH
/usr/local/opt/make/libexec/gnubin /usr/local/bin /usr/local/sbin /Users/miki725/.fzf/bin /usr/bin /bin /usr/sbin /sbin

❯ make --version
GNU Make 4.3
Built for x86_64-apple-darwin19.2.0
Copyright (C) 1988-2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

❯ make <TAB>
```

in addition it does not seem to work in arch linux where I dont monkey with the `PATH`:

```
❯ uname -a
Linux venus 5.8.6-arch1-1 #1 SMP PREEMPT Thu, 03 Sep 2020 17:27:39 +0000 x86_64 GNU/Linux

❯ make --version
GNU Make 4.3
Built for x86_64-pc-linux-gnu
Copyright (C) 1988-2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

❯ fish --version
fish, version 3.1.2

❯ make <TAB>
```

ran your script on arch:

```
❯ if make --version 2>/dev/null | string match -q 'GNU*'
                make -pRrq :
        else
                make  -d g1 -rn
        end
make: *** No rule to make target ':'.  Stop.
# GNU Make 4.3
# Built for x86_64-pc-linux-gnu
# Copyright (C) 1988-2020 Free Software Foundation, Inc.
# License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
# This is free software: you are free to change and redistribute it.
# There is NO WARRANTY, to the extent permitted by law.

# Make data base, printed on Fri Sep 11 13:31:10 2020

# Variables

# environment
SSH_TTY = /dev/pts/0
# environment
GPG_TTY = /dev/pts/0
# environment
NVM_DIR = /root/.nvm
# default
MAKE_COMMAND := make
# automatic
@D = $(patsubst %/,%,$(dir $@))
# default
.VARIABLES :=
# environment
PWD = /tmp/tmp.HVzhErM2V8
# automatic
%D = $(patsubst %/,%,$(dir $%))
# environment
LSCOLORS = dxfxcxdxbxegedabagacad
# environment
MAIL = /var/spool/mail/root
# automatic
^D = $(patsubst %/,%,$(dir $^))
# automatic
%F = $(notdir $%)
# environment
LANG = en_US.UTF-8
# default
.LOADED :=
# default
.INCLUDE_DIRS = /usr/include /usr/local/include /usr/include
# makefile
MAKEFLAGS = pqrR
# makefile
CURDIR := /tmp/tmp.HVzhErM2V8
# automatic
*D = $(patsubst %/,%,$(dir $*))
# environment
N_PREFIX = /root/.n
# environment
MFLAGS = -pqrR
# environment
SSH_AUTH_SOCK = /run/user/0/gnupg/S.gpg-agent.ssh
# default
.SHELLFLAGS := -c
# automatic
+D = $(patsubst %/,%,$(dir $+))
# makefile (from 'Makefile', line 1)
MAKEFILE_LIST := Makefile
# automatic
@F = $(notdir $@)
# environment
XDG_SESSION_TYPE = tty
# automatic
?D = $(patsubst %/,%,$(dir $?))
# automatic
*F = $(notdir $*)
# environment
DBUS_SESSION_BUS_ADDRESS = unix:path=/run/user/0/bus
# automatic
<D = $(patsubst %/,%,$(dir $<))
# default
MAKE_HOST := x86_64-pc-linux-gnu
# environment
ANDROID_HOME = /usr/local/opt/android-sdk
# makefile
SHELL = /bin/sh
# default
MAKECMDGOALS := :
# environment
SHLVL = 2
# environment
MAKELEVEL := 0
# default
MAKE = $(MAKE_COMMAND)
# environment
PATH = /usr/local/bin:/usr/local/sbin:/root/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl
# default
MAKEFILES :=
# environment
MOTD_SHOWN = pam
# automatic
^F = $(notdir $^)
# environment
SSH_CLIENT = 10.10.50.101 50316 22
# automatic
?F = $(notdir $?)
# automatic
+F = $(notdir $+)
# 'override' directive
GNUMAKEFLAGS :=
# environment
LOGNAME = root
# makefile
.DEFAULT_GOAL := foo
# environment
EDITOR = nvim
# environment
USER = root
# default
MAKE_VERSION := 4.3
# environment
STARSHIP_SHELL = fish
# environment
SSH_AGENT_PID =
# default
MAKE_TERMOUT := /dev/pts/0
# environment
XDG_RUNTIME_DIR = /run/user/0
# default
MAKE_TERMERR := /dev/pts/0
# environment
XDG_SESSION_CLASS = user
# environment
HOME = /root
# environment
TERM = xterm-256color
# environment
XDG_SESSION_ID = 76
# default
.RECIPEPREFIX :=
# environment
RIPGREP_CONFIG_PATH = /root/.config/ripgrep/.ripgreprc
# automatic
<F = $(notdir $<)
# default
SUFFIXES :=
# default
.FEATURES := target-specific order-only second-expansion else-if shortest-stem undefine oneshell nocomment grouped-target extra-prereqs archives jobserver output-sync check-symlink guile load
# environment
SSH_CONNECTION = 10.10.50.101 50316 10.10.20.20 22
# variable set hash-table stats:
# Load=65/1024=6%, Rehash=0, Collisions=4/97=4%

# Pattern-specific Variable Values

# No pattern-specific variable values.

# Directories

# . (device 48, inode 44915034): 3 files, no impossibilities.

# 3 files, no impossibilities in 1 directories.

# Implicit Rules

baz%:
#  recipe to execute (from 'Makefile', line 6):
	echo $*

# 1 implicit rules, 0 (0.0%) terminal.
# Files

# Not a target:
Makefile:
#  Implicit rule search has been done.
#  Last modified 2020-09-11 13:28:10.273805503
#  File has been updated.
#  Successfully updated.

# Not a target:
::
#  Command line target.
#  Implicit rule search has been done.
#  File does not exist.
#  File has not been updated.

# Not a target:
.DEFAULT:
#  Implicit rule search has not been done.
#  Modification time never checked.
#  File has not been updated.

bar:
#  Implicit rule search has not been done.
#  Modification time never checked.
#  File has not been updated.
#  recipe to execute (from 'Makefile', line 4):
	echo bar

foo:
#  Implicit rule search has not been done.
#  Modification time never checked.
#  File has not been updated.
#  recipe to execute (from 'Makefile', line 2):
	echo foo

# files hash-table stats:
# Load=6/1024=1%, Rehash=0, Collisions=0/16=0%
# VPATH Search Paths

# No 'vpath' search paths.

# No general ('VPATH' variable) search path.

# strcache buffers: 1 (0) / strings = 11 / storage = 80 B / avg = 7 B
# current buf: size = 8162 B / used = 80 B / count = 11 / avg = 7 B

# strcache performance: lookups = 17 / hit rate = 35%
# hash-table stats:
# Load=11/8192=0%, Rehash=0, Collisions=0/17=0%
# Finished Make data base on Fri Sep 11 13:31:10 2020
```

not sure what change in gnu make 4 to break this. let me know if I can give any other info to help to debug this

**Comment by faho** [comment]
Ah, okay. Turns out this was fixed in 79566bc32544b147452c2cad6a51b903ef6d7387, to be included in the next release.

**Comment by miki725** [comment]
awesome. thanks for such quick response!
